app/Http/Controllers/PostController.php

```diff
@@ -1,18 +1,49 @@
 <?php
 
 namespace App\Http\Controllers;
+use App\Location;
+use App\Post;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
 
 class PostController extends Controller
 {
 
-    public function create()
+    public function create($id)
     {
-        return view('posts.create');
+        //$location = Location::with('posts')->findOrFail($id);
+
+        return view('posts.create')->with("location", $id);
     }
 
-    public function store()
+    public function store(Request $request, $id)
```

Comment:
Put actual arguments _before_ typehinted dependencies.

So make it `store($id, Request $request)`.

It makes it clear what are taken as arguments by putting them first.